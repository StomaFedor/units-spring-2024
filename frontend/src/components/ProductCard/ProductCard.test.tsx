import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { ProductCard } from './ProductCard';
import {Product} from "../../types";

const testProduct: Product = {
    id: 1,
    name: '1',
    description: '1',
    price: 1,
    priceSymbol: '$',
    category: 'Электроника',
    imgUrl: '1.png',
};

const testProductNoImg: Product = {
    id: 2,
    name: '2',
    description: '2',
    price: 2,
    priceSymbol: '$',
    category: 'Электроника',
};

describe('ProductCard test', () => {

    it('should render correctly with image', () => {
        const rendered = render(<ProductCard {...testProduct} />);
        expect(rendered.asFragment()).toMatchSnapshot();
    });

    it('should render correctly without image', () => {
        const rendered = render(<ProductCard {...testProductNoImg} />);
        expect(rendered.asFragment()).toMatchSnapshot();
    });
});