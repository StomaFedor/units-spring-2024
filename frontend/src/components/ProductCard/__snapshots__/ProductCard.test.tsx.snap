// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCard test should render correctly with image 1`] = `
<DocumentFragment>
  <div
    class="product-card"
  >
    <div
      class="product-card__text"
    >
      <h2
        class="product-card__name"
      >
        1
      </h2>
      <div
        class="product-card__description"
      >
        1
      </div>
      <h1
        class="product-card__price"
      >
        1 $
      </h1>
      <div
        class="product-card__category"
      >
        Электроника
      </div>
    </div>
    <img
      alt="1"
      class="product-card__image"
      src="1.png"
    />
  </div>
</DocumentFragment>
`;

exports[`ProductCard test should render correctly without image 1`] = `
<DocumentFragment>
  <div
    class="product-card"
  >
    <div
      class="product-card__text"
    >
      <h2
        class="product-card__name"
      >
        2
      </h2>
      <div
        class="product-card__description"
      >
        2
      </div>
      <h1
        class="product-card__price"
      >
        2 $
      </h1>
      <div
        class="product-card__category"
      >
        Электроника
      </div>
    </div>
  </div>
</DocumentFragment>
`;
